ChemistryLib/PhreeqcIO.cpp

```diff
@@ -301,6 +323,41 @@ std::ostream& operator<<(std::ostream& os, PhreeqcIO const& phreeqc_io)
             os << "-steps " << phreeqc_io._dt << "\n" << "\n";
         }
 
+        os << "USE solution none" << "\n";
+        os << "END" << "\n" << "\n";
+
+        auto const& surface = phreeqc_io._surface;
+        if (!surface.empty())
+        {
+                os << "SURFACE " << chemical_system_id + 1 << "\n";
+                std::size_t aqueous_solution_id =
+                    dump->aqueous_solutions_prev.empty()
+                        ? chemical_system_id + 1
+                        : num_chemical_systems + chemical_system_id + 1;
+                os << "-equilibrate with solution " << aqueous_solution_id << "\n";
+                os << "-sites_units DENSITY" << "\n";
```

Comment:
if you add a variable for the absolute amount of surface *sites*, defining the *-sites_units* here will be optional.
You need to add it only in case you define a surface using *site_density* in `CreateSurface.cpp`.

Reply:
I prefer to keep the implementation as it is in this pull request, and set up another pull request  for this extension. @joboog

Reply:
ok.